automate.py

```diff
@@ -164,6 +167,60 @@ def run_important_traces(self):
                 event.content['message'] = content
                 self._tracer.controller.buffers.app.put(event)
 
+    def schedule_id(self, id_):
+        """Keep track of scheduled ids"""
+        if not isinstance(id_, str):
+            raise AttributeError("Invalid id type, for schedule.")
+        self.ids.add(id_)
+        return id_
+
+    def schedule_traces(self, func, settings_=settings):
+        """Check for invalid arguments from schedule"""
+        if settings_.TRIGGER_SCHEDULE_TRACES:
+            id_ = self.schedule_id('automatic_traces')
+            trigger = settings_.SCHEDULE_TRIGGER
+            kwargs = settings_.SCHEDULE_ARGS
+            if (not isinstance(kwargs, dict) or
+                    not isinstance(trigger, str)):
+                raise AttributeError("Invalid attributes for "
+                                     "job to be scheduled.")
+            trigger_args = {}
+            trigger_args['kwargs'] = {'trigger': trigger}
+            trigger_args['kwargs'].update(kwargs)
+            self.scheduler.add_callable(id_, func, **trigger_args['kwargs'])
```

Comment:
If on `Scheduler.add_callable` you return the job instance you can get rid of this extra call below `get_job`, simplifying the code. Could you update this, Aldo?

Reply:
Yes, will do.